packages/components/src/Select/Select.tsx

```diff
@@ -1,30 +1,185 @@
+import { useFocus } from '@react-aria/interactions';
+import { SelectState, useSelectState } from '@react-stately/select';
+import { useButton } from '@react-aria/button';
+import { SelectProps } from '@react-types/select';
+
+import { PolymorphicComponent, PolymorphicProps } from '@marigold/types';
 import { ArrowDown } from '@marigold/icons';
 import { useStyles } from '@marigold/system';
-import { ComponentProps } from '@marigold/types';
-import { Box } from '../Box';
 
-export type SelectProps = { variant?: string } & ComponentProps<'select'>;
+import { Box, BoxOwnProps } from '../Box';
+import { Text } from '../Text';
+
+export type SelectOwnProps = {
+  variant?: string;
+  label?: string;
+  disabled?: boolean;
+} & BoxOwnProps &
+  SelectProps<any>;
+export type SelectPolymorphicProps = PolymorphicProps<SelectOwnProps, 'select'>;
 
-export const Select: React.FC<SelectProps> = ({
+export const Select = (({
   variant = 'default',
-  children,
+  label,
+  disabled,
   ...props
-}) => {
-  const iconStyles = useStyles({
-    css: {
-      alignSelf: 'center',
-      pointerEvents: 'none',
-      ml: '-28px',
-    },
-  });
+}: SelectPolymorphicProps) => {
+  const state = useSelectState(props);
+  const ref = useRef<any>();
+  const { labelProps, triggerProps, valueProps, menuProps } = useSelect(
+    { ...props },
+    state,
+    ref
+  );
+  let { buttonProps } = useButton(triggerProps, ref);
+  // const iconStyles = useStyles({
+  //   css: {
+  //     alignSelf: 'center',
+  //     pointerEvents: 'none',
+  //   },
+  // });
 
   return (
-    <Box display="flex">
-      <Box as="select" variant={`select.${variant}`} {...props}>
-        {children}
+    <Box position="relative" display="inline-block">
+      <Box {...labelProps}>
+        <Text>{label}</Text>
       </Box>
-      <ArrowDown className={iconStyles} />
+      <HiddenSelect
+        state={state}
+        triggerRef={ref}
+        label={label}
+        name={props.name}
+      />
+      <button
+        {...buttonProps}
+        ref={ref}
+        style={{
+          fontFamily: 'Inter',
+          fontSize: '14px',
+          fontWeight: 400,
+          lineHeight: '32px',
+          background: '#FFFFFF',
+          border: '1px solid #CCCCCC',
+          borderRadius: '2px',
+        }}
+      >
+        <Box as="span" {...valueProps}>
+          {state.selectedItem
+            ? state.selectedItem.rendered
+            : 'Select an option'}
+        </Box>
+        <Box as="span" aria-hidden pl="5px">
+          {/* <ArrowDown className={iconStyles} /> */}▼
+        </Box>
+      </button>
+      {state.isOpen && <ListBoxPopup {...menuProps} state={state} />}
     </Box>
   );
+}) as PolymorphicComponent<SelectOwnProps, 'select'>;
+
+// ListBox
+type ListBoxPopupProps = {
+  state: SelectState<any>;
+  otherProps?: React.HTMLAttributes<HTMLElement>;
+};
+
+const ListBoxPopup = ({ state, ...otherProps }: ListBoxPopupProps) => {
```

Comment:
Maybe put that in a separate file for better overview? :)

Reply:
Yes good Idea. But where? It is a good place between the other "normal" components?

Reply:
If it's only usable die the select, or it in the select folder and don't expose it